workitem/workitem_repository.go

```diff
@@ -701,50 +702,182 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
 	return &result, nil
 }
 
-// GetCountsPerIteration fetches WI count from DB and returns a map of iterationID->WICountsPerIteration
-// This function executes following query to fetch 'closed' and 'total' counts of the WI for each iteration in given spaceID
-// 	SELECT iterations.id as IterationId, count(*) as Total,
-// 		count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed
-// 		FROM "work_items" left join iterations
-// 		on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb
-// 		WHERE (iterations.space_id = '33406de1-25f1-4969-bcec-88f29d0a7de3'
-// 		and work_items.deleted_at IS NULL) GROUP BY IterationId
+// GetCountsPerIteration counts WIs including iteration-children and returns a map of iterationID->WICountsPerIteration
 func (r *GormWorkItemRepository) GetCountsPerIteration(ctx context.Context, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
-	defer goa.MeasureSince([]string{"goa", "db", "workitem", "getCountPerIteration"}, time.Now())
+	defer goa.MeasureSince([]string{"goa", "db", "workitem", "getCountsPerIteration"}, time.Now())
+	db := r.db.Model(&iteration.Iteration{}).Where("space_id = ?", spaceID)
+	if db.Error != nil {
+		return nil, errors.NewInternalError(ctx, db.Error)
+	}
+	var allIterations []uuid.UUID
+	db.Pluck("id", &allIterations)
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	iterationWithWICount := fmt.Sprintf(`
+	SELECT count(*) AS Total,
+		count(CASE fields->>'system.state'
+					WHEN 'closed' THEN '1'
+					ELSE NULL
+				END) AS Closed,
+		fields->>'system.iteration' AS iterationID
+	FROM %s wi
+	WHERE fields->>'system.iteration' IN
+		(SELECT id::text
+		FROM %s
+		WHERE space_id=?)
+	AND wi.deleted_at IS NULL
+	GROUP BY fields->>'system.iteration'`,
+		workitemTableName,
+		iterationTableName)
+	db = r.db.Raw(iterationWithWICount, spaceID.String())
 	var res []WICountsPerIteration
-	db := r.db.Table("work_items").Select(`iterations.id as IterationId, count(*) as Total,
-				count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed`).Joins(`left join iterations
-				on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb`).Where(`iterations.space_id = ?
-				and work_items.deleted_at IS NULL`, spaceID).Group(`IterationId`).Scan(&res)
+	db.Scan(&res)
 	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"space_id": spaceID.String(),
```

Comment:
thanks removed in https://github.com/almighty/almighty-core/pull/1151/commits/da355308bc11722e77d6b67ba528803219a05d70